CMakeLists.txt

```diff
@@ -27,64 +27,203 @@
 cmake_minimum_required(VERSION 2.8.11)
 
 project(dropt)
+set(PROJECT_VERSION 2.0.1)
+set(PROJECT_DESCRIPTION "dropt is yet another C and C++ library for parsing command-line options.")
+set(PROJECT_HOMEPAGE_URL "https://github.com/jamesderlin/dropt")
+
+option(BUILD_SHARED_LIBS "Build shared libraries" ON)
+
+include(GNUInstallDirs)
+include(CPackComponent)
+include(CMakePackageConfigHelpers)
+
+set(PackagingTemplatesDir "${CMAKE_CURRENT_SOURCE_DIR}/packaging")
+set(IncludeDir "${CMAKE_CURRENT_SOURCE_DIR}/include")
+set(SrcDir "${CMAKE_CURRENT_SOURCE_DIR}/src")
+
+function(configure_pkgconfig_file PKG_CONFIG_NAME componentName)
+	set(PKG_CONFIG_FILE_NAME "${CMAKE_CURRENT_BINARY_DIR}/${PKG_CONFIG_NAME}.pc")
+	configure_file("${PackagingTemplatesDir}/pkgconfig.pc.in" "${PKG_CONFIG_FILE_NAME}" @ONLY)
+	install(FILES "${PKG_CONFIG_FILE_NAME}"
+		DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
+		COMPONENT ${componentName}
+	)
+endfunction()
+
+set(${PROJECT_NAME}_h_files
+    ${IncludeDir}/${PROJECT_NAME}.h
+    ${IncludeDir}/${PROJECT_NAME}_string.h
+)
 
-include_directories(${CMAKE_CURRENT_LIST_DIR}/include)
+set(${PROJECT_NAME}_c_files
+    ${SrcDir}/${PROJECT_NAME}.c
+    ${SrcDir}/${PROJECT_NAME}_handlers.c
+    ${SrcDir}/${PROJECT_NAME}_string.c
+)
 
-set(dropt_h_files
-    ./include/dropt.h
-    ./include/dropt_string.h
+add_library(${PROJECT_NAME}
+    ${${PROJECT_NAME}_c_files}
+    ${${PROJECT_NAME}_h_files}
 )
 
-set(dropt_c_files
-    ./src/dropt.c
-    ./src/dropt_handlers.c
-    ./src/dropt_string.c
+target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include> $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+
+cpack_add_component(c_lib
+	DISPLAY_NAME "${PROJECT_NAME}"
+	DESCRIPTION "The C library"
+)
+cpack_add_component(c_lib_dev
+	DISPLAY_NAME "${PROJECT_NAME} SDK"
+	DESCRIPTION "SDK for the C library"
+)
+set(CPACK_DEBIAN_C_LIB_PACKAGE_NAME "lib${PROJECT_NAME}")
+
+set(CPACK_DEBIAN_C_LIB_DEV_PACKAGE_NAME "${CPACK_DEBIAN_C_LIB_PACKAGE_NAME}-dev")
+set(CPACK_DEBIAN_C_LIB_DEV_PACKAGE_SUGGESTS "cmake, pkg-config, pkg-conf")
+set(CPACK_DEBIAN_C_LIB_DEV_PACKAGE_DEPENDS "${CPACK_DEBIAN_C_LIB_PACKAGE_NAME}")
+
+set(CPACK_RPM_C_LIB_PACKAGE_NAME "${CPACK_DEBIAN_C_LIB_PACKAGE_NAME}")
+
+set(CPACK_RPM_C_LIB_DEV_PACKAGE_NAME "${CPACK_RPM_C_LIB_PACKAGE_NAME}-devel")
+set(CPACK_RPM_C_LIB_DEV_PACKAGE_SUGGESTS "${CPACK_DEBIAN_PACKAGE_SUGGESTS}")
```

Comment:
Is there a relationship between `CPACK_DEBIAN_PACKAGE_SUGGESTS` and `CPACK_DEBIAN_C_LIB_DEV_PACKAGE_SUGGESTS`?

Reply:
Fixed.